operations/jobs.go

```diff
@@ -0,0 +1,166 @@
+	var err error
+
+	go func() {
+		<-ctx.Done()
+		uo.reqCtxCancel()
+	}()
+
+	if _, err = repository.Update(uo.reqCtx, uo.object, uo.labelChanges, uo.criteria...); err != nil {
+		log.D().Debugf("Failed to execute UPDATE operation with id (%s) for entity %s", uo.operationID, uo.object.GetType())
+
+		if opErr := updateOperationState(uo.reqCtx, repository, uo.operationID, types.FAILED, &OperationError{Message: err.Error()}); opErr != nil {
+			log.D().Debugf("Failed to set state of operation with id (%s) to %s", uo.operationID, types.FAILED)
+			err = fmt.Errorf("%s : %s", err, opErr)
+		}
+		return uo.operationID, err
+	}
+
+	log.D().Debugf("Successfully executed UPDATE operation with id (%s) for entity %s", uo.operationID, uo.object.GetType())
+	if err = updateOperationState(uo.reqCtx, repository, uo.operationID, types.SUCCEEDED, nil); err != nil {
+		log.D().Debugf("Failed to set state of operation with id (%s) to %s", uo.operationID, types.SUCCEEDED)
+	}
+
+	return uo.operationID, err
+}
+
+// Execute executes a Delete DB operation
+func (do *DeleteJob) Execute(ctx context.Context, repository storage.Repository) (string, error) {
+	var err error
+
+	go func() {
+		<-ctx.Done()
+		do.reqCtxCancel()
+	}()
+
+	if err = repository.Delete(do.reqCtx, do.objectType, do.criteria...); err != nil {
+		log.D().Debugf("Failed to execute DELETE operation with id (%s) for entity %s", do.operationID, do.objectType)
+
+		if opErr := updateOperationState(do.reqCtx, repository, do.operationID, types.FAILED, &OperationError{Message: err.Error()}); opErr != nil {
+			log.D().Debugf("Failed to set state of operation with id (%s) to %s", do.operationID, types.FAILED)
+			err = fmt.Errorf("%s : %s", err, opErr)
+		}
+		return do.operationID, err
+	}
+
+	log.D().Debugf("Successfully executed DELETE operation with id (%s) for entity %s", do.operationID, do.objectType)
+	if err = updateOperationState(do.reqCtx, repository, do.operationID, types.SUCCEEDED, nil); err != nil {
+		log.D().Debugf("Failed to set state of operation with id (%s) to %s", do.operationID, types.SUCCEEDED)
+	}
+
+	return do.operationID, err
+}
+
+func updateOperationState(ctx context.Context, repository storage.Repository, operationID string, state types.OperationState, opErr *OperationError) error {
+	operation, err := fetchOperation(ctx, repository, operationID)
+	if err != nil {
+		return err
+	}
+
+	operation.State = state
+	operation.UpdatedAt = time.Now()
+
+	if opErr != nil {
+		bytes, err := json.Marshal(opErr)
+		if err != nil {
+			return err
+		}
+		operation.Errors = json.RawMessage(bytes)
+	}
+
+	_, err = repository.Update(ctx, operation, query.LabelChanges{})
+	if err != nil {
+		log.D().Debugf("Failed to update state of operation with id (%s) to SUCCEEDED", operationID)
+		return err
+	}
+
+	log.D().Debugf("Successfully updated state of operation with id (%s) to SUCCEEDED", operationID)
```

Comment:
`log.D().Debugf("Failed to update state of operation with id (%s) to %s", operationID, state)`
It is not only SUCCEEDED